nx/lib/nx.ex

```diff
@@ -7903,22 +7919,90 @@ defmodule Nx do
   @doc type: :linalg
   def triangular_solve(a, b, opts \\ []) do
     %T{shape: s1 = {m, _}} = a = tensor!(a)
-    %T{shape: {q}} = b = tensor!(b)
+    %T{shape: b_shape} = b = tensor!(b)
 
     case shape(s1) do
       {n, n} -> {n, n}
       other -> raise ArgumentError, "expected a square matrix, got: #{inspect(other)}"
     end
 
-    if m != q do
-      raise ArgumentError, "incompatible dimensions for a and b on tringular solve"
+    case b_shape do
+      {^m, ^m} ->
+        nil
+
+      {^m} ->
+        nil
+
+      _ ->
+        raise ArgumentError, "incompatible dimensions for a and b on tringular solve"
```

Comment:
I believe this should be "triangular" and not "tringular"

Reply:
```suggestion
        raise ArgumentError, "incompatible dimensions for a and b on triangular solve"
```

Reply:
yeap